### DISCUSSION TITLE
[Photos desktop] Improve performance for > 1TB Google Takeout imports

### DISCUSSION THREAD

**Discussion by mnvr** [ente-io/ente#3420] [Enhancements]
Some people have reported issues when trying to import Google Takeouts > 1TB. Need to investigate, it is possible that there is some underlying performance issue of using some inappropriate data structure somewhere in the import flow.

For people hitting this, one workaround might be to import in batches.

There is also related issue with the app running out of memory when importing many large multi-GB videos. The people doing > 1TB imports usually have many large multi-GB videos, so they often hit this issue, but this is likely a distinct problem (possibly related to current 4GB [Electron](https://www.electronjs.org/blog/v8-memory-cage) app memory limit). Workaround is to upload these offending videos [separately, or via the web app ](https://help.ente.io/photos/troubleshooting/large-uploads)

**Comment by keithbrink** [comment]
Personally I would like to see more uploads in parallel, I think 4 is too small. Configurable would be nice.

**Reply by ahxxm** [reply]
found an exact same request in #4271

**Comment by mariolameiras** [comment]
I am having the same problem. It has taken me an entire day to upload just 50Gb and I have 500Gb to go.   Any ideas on how I could speed up?

Would it help to disable Machine Learning?

**Reply by vishnukvmd** [reply]
Machine learning does not affect upload speeds.

Could you please run a speed test against `EU Central` and let us know if you notice an issue? https://www.backblaze.com/cloud-backup/resources/speedtest

**Reply by mariolameiras** [reply]
Without stopping my currently upload process in ente.io I get: 
![image](https://github.com/user-attachments/assets/7accaede-9c98-43e0-a450-b8a6ca11825f)

My current estimated upload rate using ente.io upload is 200Gb/day, way less than the 3000Gb estimated in that page. 

I have also tried uploading an unzipped folders instead of a zip but I saw no differences in speed

**Reply by vishnukvmd** [reply]
Could you share logs (Help > View logs) with support@ente.io? This will help us understand what the bottleneck is. Thanks!

**Comment by ahxxm** [comment]
I've around ~10,000 photos, it took 4-5 hours to upload.

My observation is that resource usages aren't pipelined, the CPU and network have spikes instead of consistent load, and upload progress/percentages seem to support this idea.

For each new entry in "Upload in progress":
- quickly reaches to 90%+ => CPU and network, or encryption and upload speed, aren't bottleneck
- then stucks for a while => perhaps waiting for server/db commit? this looks suspicious
- then succeeds and disappeares, and another entry starts to repeat the process

TCPView showed that ente.exe connects to a nearby Cloudflare IP, re-uses 1 TCP connection -- as local port is fixed -- to upload.

**Reply by ahxxm** [reply]
Instead of 4 whole tasks of encrypt+upload+commit, I think splitting to 4 encrypt tasks + 4 active upload tasks might make upload much faster, latter might also need some server-side work and could bring extra complexity

**Comment by AayushKucheria** [comment]
Any updates on this? I haven't moved the bulk of my data from google photos because it just takes so long, and I'd rather wait for this to be solved before.
